### DISCUSSION TITLE
Does Bull Flush the redis cache occasionally ?

### DISCUSSION THREAD

**Discussion by JohnAtPrelude** [OptimalBits/bull#2393] [Q&A] [answered]
Hi,
I'm tracking an issue with my redis installation.
Sometimes, suddenly and with no apparent reasoning my entire redis cache is flushed (per everything is deleted by name).
I have checked in the redis stats and nothing obvious jumps out at me.

I share this redis installation with some other code, so both bull and my code are writing to the same redis, but my items are named by a different schema.

Im curious is there a mechanism somewhere in bull that will either flush the entire redis cache, or just delete everything, including things that are not bull managed items?

**Comment by manast** [comment]
Can you check that your maxmemory policy in your Redis install is set to no-eviction?

Manuel Astudillo
CEO Castmill AB
https://castmill.com <https://castmill.com> 




On 4 Jul 2022, at 02:13, Prelude Pro ***@***.*** ***@***.***> > wrote:


Hi,
I'm tracking an issue with my redis installation.
Sometimes, suddenly and with no apparent reasoning my entire redis cache is flushed (per everything is deleted by name).
I have checked in the redis stats and nothing obvious jumps out at me.

I share this redis installation with some other code, so both bull and my code are writing to the same redis, but my items are named by a different schema.

Im curious is there a mechanism somewhere in bull that will either flush the entire redis cache, or just delete everything, including things that are not bull managed items?

**Comment by JohnAtPrelude** [comment]
> Hi, I'm tracking an issue with my redis installation. Sometimes, suddenly and with no apparent reasoning my entire redis cache is flushed (per everything is deleted by name). I have checked in the redis stats and nothing obvious jumps out at me.
> 
> I share this redis installation with some other code, so both bull and my code are writing to the same redis, but my items are named by a different schema.
> 
> Im curious is there a mechanism somewhere in bull that will either flush the entire redis cache, or just delete everything, including things that are not bull managed items?

Yes, my maxmemorypolicy is set to noeviction
I'm using a vanilla install of the redis docker image

Thanks

**Comment by manast** [comment]
The only place where Bull flushes Redis is when running its tests, and then only on the local redis instance.

**Comment by JohnAtPrelude** [comment, marked as answer]
I just wanted to follow up on this for the sake of anyone else who may be dealing with this.

It turned out that my redis instance was getting hacked.
I was an idiot and had not properly secured my Security Group in my Firewall and the redis installation was wide open to the public.
There were some malicious players out there port scanning and looking for redis vulnerabilities who found my server and were wrecking it on a fairly frequent basis.

I ended up moving my redis instance off of the detalut port, closing the hole in the firewall and enforcing basic password authentication on redis queries.

**Reply by manast** [reply]
If you leave any system unprotected without user/password it will be hacked, thats 100% guaranteed. Please be careful.
Here something I wrote some time ago: https://docs.taskforce.sh/security-primer
